### PR TITLE
Instrument jdbc datasources with OpenTelemetry

### PR DESCRIPTION
**Pull request by cyrille-leclerc**
Instrument jdbc datasources with OpenTelemetry

### Testing done

End to end tests:

* Database Plugin + Junit SQL Storage Plugin doesn't enable the OpenTelemetry JDBC instrumentation
* Database Plugin + Junit SQL Storage Plugin + configured OTel Plugin doesn't enable the OpenTelemetry JDBC instrumentation
* Database Plugin + Junit SQL Storage Plugin + configured OTel Plugin with `otel.instrumentation.jdbc.enabled=true` enables OTel JDBC instrumentation on the Jenkins Controller but not on the build agents
* Database Plugin + Junit SQL Storage Plugin + configured OTel Plugin with `otel.instrumentation.jdbc.enabled=true` & `otel.instrumentation.jenkins.agent.enabled=true` enables OTel JDBC instrumentation both on the Jenkins Controller nd on the build agents. 

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,6 +113,22 @@ public class TestRow {
 }
 ```
 
+# Troubleshooting performance problems with OpenTelemetry and tracing
```
Comment (timja) [inline README.md:116]:
```suggestion
## Troubleshooting performance problems with OpenTelemetry and tracing
```

**Review by timja — COMMENTED** [review]
LGTM untested, could you fill in the testing done section of the template please? to cover off what types of testing you've done? e.g. controller, agent etc.

**Comment by cyrille-leclerc** [comment]
Thanks @timja , details on all the tests added in the PR description (end to end tests)

**Comment by timja** [comment]
Thanks!

**Pull request merged**